_preload.lua

```diff
@@ -51,62 +53,38 @@
 		kind = "integer",
 	}
 
-	if _ACTION >= "vs2015" then
-		api.register {
-			name = "toolchainversion",
-			scope = "config",
-			kind = "string",
-			allowed = {
-				"4.9", -- NDK GCC versions
-				"3.6", -- NDK clang versions
-			},
-		}
-	else
-		api.register {
-			name = "toolchainversion",
-			scope = "config",
-			kind = "string",
-			allowed = {
-				"4.6", -- NDK GCC versions
-				"4.8",
-				"4.9",
-				"3.4", -- NDK clang versions
-				"3.5",
-				"3.6",
-			},
-		}
-	end
+	api.register {
+		name = "toolchainversion",
+		scope = "config",
+		kind = "string",
+		allowed = {
+			"4.6", -- NDK GCC versions
+			"4.8",
+			"4.9",
+			"3.4", -- NDK clang versions
+			"3.5",
+			"3.6",
+			"3.8",
+		},
+	}
 
-	if _ACTION >= "vs2015" then
-		api.register {
-			name = "stl",
-			scope = "config",
-			kind = "string",
-			allowed = {
-				"minimal c++ (system)",
-				"c++ static",
-				"c++ shared",
-				"stlport static",
-				"stlport shared",
-				"gnu stl static",
-				"gnu stl shared",
-				"llvm libc++ static",
-				"llvm libc++ shared",
-			},
-		}
-	else
-		api.register {
-			name = "stl",
-			scope = "config",
-			kind = "string",
-			allowed = {
-				"none",
-				"minimal",
-				"stdc++",
-				"stlport",
-			},
-		}
-	end
+	api.register {
+		name = "stl",
+		scope = "config",
+		kind = "string",
+		allowed = {
```

Comment:
I suggest looking at how I redid this in my PR. I don't know if it's the correct path forward, but it could be a better alternative than what we've currently got?